packages/common/src/audius-query/createApi.ts

```diff
@@ -279,7 +294,23 @@ const fetchData = async <Args, Data>(
 
     endpoint.onQueryStarted?.(fetchArgs, { dispatch })
 
-    const apiData = await endpoint.fetch(fetchArgs, context)
+    // If endpoint config specifies retries wrap the fetch
+    // and return a single error at the end if all retries fail
+    const apiData = endpoint.options.retry
+      ? await retry(
+          async (bail) => {
+            try {
+              return endpoint.fetch(fetchArgs, context)
+            } catch (e) {
+              if (isNonRetryableError(e)) {
+                bail(new Error(`Non-retryable error: ${e}`))
+              }
+              return null
+            }
+          },
+          { ...defaultRetryConfig, ...endpoint.options.retryConfig }
```

Comment:
does `retry` have a default delay? What would make a retry succeed when the initial request failed?

Reply:
Looking at the docs within docs: https://github.com/tim-kos/node-retry#retrytimeoutsoptions
The default `minTimeout` is 1000 ms, and based on the formula documented, the retries would occur at intervals of 2000, 4000, 8000ms, with some randomization (this is confusing because the vercel docs say randomization is on by default, while the inner library says it off by default).

I can definitely adjust this if it feels like it takes too long. The video in the description is a good example of how long it would take in the worst case.